website/src/components/Chat/ChatAssistantDraftViewer.tsx

```diff
@@ -0,0 +1,234 @@
+    setIsLoading(areAllMessagesEmpty);
+  }, [streamedDrafts, setIsLoading]);
+
+  useEffect(() => {
+    const areAllMessagesComplete =
+      regenIndex < draftMessages.length &&
+      draftMessages[regenIndex]?.length !== 0 &&
+      draftMessages[regenIndex].every((message) =>
+        ["complete", "aborted_by_worker", "cancelled", "timeout"].includes(message.state)
+      );
+    setIsComplete(areAllMessagesComplete);
+  }, [regenIndex, draftMessages, setIsComplete]);
+
+  const handlePrevious = useCallback(() => {
+    setRegenIndex(regenIndex > 0 ? regenIndex - 1 : regenIndex);
+  }, [setRegenIndex, regenIndex]);
+
+  const handleNext = useCallback(() => {
+    setRegenIndex(regenIndex < draftMessages.length - 1 ? regenIndex + 1 : regenIndex);
+  }, [setRegenIndex, regenIndex, draftMessages]);
+
+  useEffect(() => {
+    if (!draftMessages || draftMessages.length === 0) {
+      setRegenIndex(0);
+    }
+  }, [draftMessages, setRegenIndex]);
+
+  const handleToggleExpand = (index) => {
+    setExpandedMessage(index === expandedMessage ? -1 : index);
+  };
+
+  const handleRetry = useCallback(() => {
+    if (onRetry && regenIndex < draftMessages.length && draftMessages[regenIndex]?.length !== 0) {
+      setExpandedMessage(0);
+      onRetry({ parentId: draftMessages[regenIndex][0].parent_id, chatId: draftMessages[regenIndex][0].chat_id });
+      setRegenIndex(draftMessages.length);
+    }
+  }, [onRetry, regenIndex, draftMessages, setExpandedMessage, setRegenIndex]);
+
+  const containerBackgroundColor = useColorModeValue("#DFE8F1", "#42536B");
+  const responseHoverBackgroundColor = useColorModeValue("gray.300", "blackAlpha.300");
+
+  return (
+    <Flex
+      key="chat-message-select"
+      width="full"
+      gap={0.5}
+      flexDirection={{ base: "column", md: "row" }}
+      alignItems="start"
+      position="relative"
+      padding={{ base: 3, md: 0 }}
+      background={{ base: containerBackgroundColor, md: "transparent" }}
+      borderRadius={{ base: "18px", md: 0 }}
+      maxWidth={{ base: "3xl", "2xl": "4xl" }}
+    >
+      <Avatar
+        size={{ base: "xs", md: "sm" }}
+        marginRight={{ base: 0, md: 2 }}
+        marginTop={{ base: 0, md: `6px` }}
+        marginBottom={{ base: 1.5, md: 0 }}
+        borderColor={"blackAlpha.200"}
+        _dark={{ backgroundColor: "blackAlpha.300" }}
+        src="/images/logos/logo.png"
+      />
+      <Flex
+        background={containerBackgroundColor}
+        borderRadius={{ base: 0, md: "18px" }}
+        gap={1}
+        width={isLoading ? "fit-content" : "full"}
+        padding={{ base: 0, md: 4 }}
+        flexDirection="column"
+      >
+        {isLoading ? (
+          <CircularProgress isIndeterminate />
+        ) : (
+          <>
+            {(isComplete && regenIndex < draftMessages.length && draftMessages[regenIndex]?.length !== 0
+              ? draftMessages[regenIndex]
+              : streamedDrafts
+            ).map((message, index) => (
```

Comment:
TS is warning that the message is any, and will be error if we turn on strict mode. Extract to sub-component will make the code safer here.